service/storage_service.cc

```diff
@@ -749,7 +749,8 @@ void storage_service::handle_cdc_generation(std::optional<db_clock::time_point>
         return;
     }
 
-    if (!db::system_keyspace::bootstrap_complete() || !_sys_dist_ks.local_is_initialized()) {
+    if (!db::system_keyspace::bootstrap_complete() || !_sys_dist_ks.local_is_initialized()
+            || !_sys_dist_ks.local().started()) {
         // We still haven't finished the startup process.
```

Comment:
What if it's locally started but on some remote shard still in progress?

Really, these dependencies should be expressed statically rather than computed dynamically. Of course that's beyond the scope of this series.

Reply:
We only need the local instance (shard 0).
Btw. the other instances don't do anything anyway (see `system_distributed_keyspace::start`, the first `if`)